rp2040-hal/src/spi.rs

```diff
@@ -251,17 +252,33 @@ impl<S: State, D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<S, D, P, DS
     }
 }
 
-impl<D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<Disabled, D, P, DS> {
-    /// Create new not initialized Spi bus. Initialize it with [`.init`][Self::init]
+impl<D: SpiDevice, P: ValidSpiPinout<D>> Spi<Disabled, D, P, 8> {
+    /// Create new (not initialized) Spi bus with the default data size (8 bits).
+    ///
+    /// Initialize it with [`.init`][Self::init]
     /// or [`.init_slave`][Self::init_slave].
-    pub fn new(device: D, pins: P) -> Spi<Disabled, D, P, DS> {
+    pub fn new(device: D, pins: P) -> Spi<Disabled, D, P, 8> {
         Spi {
             device,
             pins,
             state: PhantomData,
         }
     }
 
+    /// Create new (not initialized) Spi bus with a non-default data size.
+    ///
+    /// Initialize it with [`.init`][Self::init]
+    /// or [`.init_slave`][Self::init_slave].
+    pub fn new_with_data_size<const DS: u8>(device: D, pins: P) -> Spi<Disabled, D, P, DS> {
```

Comment:
I wonder if this method should be moved down to the `impl<D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<Disabled, D, P, DS>`? It's a little bit strange to have the constructor for the bit-generic version in the 8-bit impl.

Reply:
Done. The only awkward thing is that now there is already `DS` const generic from the impl block in scope, so I had to rename it on the function level parameter. But not that big of a deal.

Reply:
I don't see any update on the branch on github. Did you perhaps forget to push it?

Regarding the function level parameter: I think you don't need it at all, you can use the parameter from the impl block:
```
impl<D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<Disabled, D, P, DS> {
  [...]
  pub fn new_with_data_size(device: D, pins: P) -> Spi<Disabled, D, P, DS> {
    [...]
  }
  [...]
}
```

Which makes me wonder: Is the `new_with_data_size` name confusing, if it's often possible to call it without specifying the data size at the call location? It's enough if the compiler can somehow infer the correct value for DS, eg. if you later store the value in some struct where the type is specified.

Reply:
True, but I feel like storing this in a wider struct or somewhere it could be inferred would be relatively uncommon? Maybe I'm mistaken though.